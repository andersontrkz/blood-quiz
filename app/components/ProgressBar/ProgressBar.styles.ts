import { StyleSheet } from 'react-native';
import { COLORS } from '@/constants';

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: 16,
    borderRadius: 50,
    backgroundColor: '#FFFFFF11',
  },
  progress: {
    height: 16,
    borderRadius: 50,
    backgroundColor: COLORS.primary,
  },
});

export default styles;
