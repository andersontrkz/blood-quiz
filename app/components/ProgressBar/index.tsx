import React from 'react';
import { View, Animated } from 'react-native';
import styles from './ProgressBar.styles';

type ProgressBarProps = {
  progress: Animated.Value;
  totalSteps: number;
};

export default function ProgressBar({ progress, totalSteps }: ProgressBarProps) {
  const progressAnimation = progress.interpolate({
    inputRange: [0, totalSteps],
    outputRange: ['0%', '100%'],
  });

  return (
    <View style={styles.container} testID="progress-bar">
      <Animated.View style={[styles.progress, { width: progressAnimation }]} />
    </View>
  );
}
