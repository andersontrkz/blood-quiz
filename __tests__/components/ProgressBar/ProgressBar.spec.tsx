import React from 'react';
import { Animated } from 'react-native';
import { render } from '@testing-library/react-native';

import ProgressBar from '@/app/components/ProgressBar';

describe('ProgressBar Component', () => {
  const progressMock = new Animated.Value(0);

  it('should render the progress bar correctly', () => {
    const { getByTestId } = render(
      <ProgressBar progress={progressMock} totalSteps={10} />
    );

    const progressBar = getByTestId('progress-bar');
    expect(progressBar).toBeTruthy();
  });

  it('should render correctly with at least 1 step', () => {
    const { getByTestId } = render(
      <ProgressBar progress={progressMock} totalSteps={1} />
    );

    const progressBar = getByTestId('progress-bar');
    expect(progressBar).toBeTruthy();
  });

  it('should render correctly with zero steps', () => {
    const { getByTestId } = render(
      <ProgressBar progress={progressMock} totalSteps={0} />
    );

    const progressBar = getByTestId('progress-bar');
    expect(progressBar).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const { toJSON } = render(
      <ProgressBar progress={progressMock} totalSteps={0} />
    );
  
    expect(toJSON()).toMatchSnapshot();
  });
});
