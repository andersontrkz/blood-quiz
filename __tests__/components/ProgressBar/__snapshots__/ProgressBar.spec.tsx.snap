// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Component should match the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF11",
      "borderRadius": 50,
      "height": 16,
      "width": "100%",
    }
  }
  testID="progress-bar"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#EE0F55",
        "borderRadius": 50,
        "height": 16,
        "width": "0%",
      }
    }
  />
</View>
`;
